src/demos/add-charts/app.js

```diff
@@ -160,31 +122,39 @@ export default class App extends Component {
     });
   }
 
+  _updateLayerSettings(settings) {
+    this.setState({settings});
+  }
+
   render() {
-    const {viewport, hoveredObject, points, settings, status, x, y} = this.state;
     return (
       <div>
-        {hoveredObject &&
-          <div style={{...tooltipStyle, left: x, top: y}}>
-            <div>{hoveredObject.id}</div>
+        {this.state.hoveredObject &&
+          <div style={{
+            ...tooltipStyle,
+            left: this.state.x,
+            top: this.state.y
+          }}>
+            <div>{this.state.hoveredObject.id}</div>
           </div>}
         <LayerControls
-          settings={settings}
-          propTypes={LAYER_CONTROLS}
-          onChange={this.updateLayerSettings.bind(this)}/>
+          settings={this.state.settings}
+          propTypes={HEXAGON_CONTROLS}
+          onChange={settings => this._updateLayerSettings(settings)}/>
         <MapGL
-          {...viewport}
+          {...this.state.viewport}
           mapStyle={MAPBOX_STYLE}
-          onViewportChange={this._onViewportChange.bind(this)}
+          onViewportChange={viewport => this._onViewportChange(viewport)}
           mapboxApiAccessToken={MAPBOX_TOKEN}>
           <DeckGLOverlay
-            viewport={viewport}
-            data={points}
-            onHover={this._onHover.bind(this)}
-            settings={settings}/>
+            viewport={this.state.viewport}
+            data={this.state.points}
+            onHover={hover => this._onHover(hover)}
+             {...this.state.settings}
```

Comment:
i'm just spreading the content of settings for concision in deckgl-overlay

Reply:
no it's not equivalent. the changes are reflected in deckgl-overlays. 

state.settings are set in layer-controls. during the tutorial we never go through the detail of what this does and how it works. that's why i'm ok with not going into too much detail in app.js